### ISSUE TITLE
Play Java Form#Field with DirectFieldAccess not binding default values

### ISSUE THREAD

**Issue by HongyiFu** [playframework/playframework#9794]
### Play Version

2.7.3

### API

Java

### Operating System

Mac OS Catalina 10.15

### JDK

openjdk version "1.8.0_232"
OpenJDK Runtime Environment (Zulu 8.42.0.21-CA-macosx) (build 1.8.0_232-b18)
OpenJDK 64-Bit Server VM (Zulu 8.42.0.21-CA-macosx) (build 25.232-b18, mixed mode)

Scala 2.13.0

### Library Dependencies

None, other than Play javaForms included by default

### Expected Behavior

Play 2.7 introduces direct field access for form binding. With this, we can bind our form `T` from request without getters/setters - provided we set the `play.data.Form.withDirectFieldAccess(boolean)` method or set the option in config.

The documentation does not say explicitly if this works for other scenarios or just in the case of binding from request? It is my expectation that it works everywhere, including this scenario: filling in default values and then displaying it in Twirl.

### Actual Behavior

After filling in default values for `T` i.e. the method `play.data.Form#fill(T)`, where `T` is our form class, we get a new form. Calling `play.data.Form.apply(String)` from our new form, we get a `Field`. The values are **not** in the `Field` if there are **no getters present**.

Implication: without getters, `Form` in twirl does not display correctly due to missing values.

### Reproducible Test Case

Assuming incoming request of param `fieldA=foo`:

```
public static class Data { 
    public String fieldA;
}


var form = formFactory.form(Data.class).bindFromRequest(request);
print(form.get().fieldA); // prints "foo"
print(form.apply("fieldA").value()); // prints "Optional[foo]" - works fine

var data = new Data();
data.fieldA = "bar"
var form2 = form.fill(data);
print(form2.get().fieldA); // prints "bar"
print(form2.apply("fieldA").value()); // prints "Optional.empty" (!)

```

Creating a getter for `fieldA` fixes this.

**Comment by mkurz** [comment]
Thanks for reporting @HongyiFu. I can confirm this is broken.
I know how to fix this already, will provide a pull request when I find more time.

**Comment by mkurz** [comment]
Fixed by #9798
